h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = Array.from(element.querySelectorAll('[data-ref="searchBarDropdownItem"]'));
+
+    var clearActiveDropdownItem = () => {
+      var activeItem = element.querySelector('.js-search-bar-dropdown-menu-item--active');
+      if (activeItem) {
+        activeItem.classList.remove('js-search-bar-dropdown-menu-item--active');
+      }
+    };
+
+    var updateActiveDropdownItem = element => {
+      clearActiveDropdownItem();
+      element.classList.add('js-search-bar-dropdown-menu-item--active');
+    };
+
+    var selectFacet = facet => {
+      this._input.value = this._input.value + facet;
+
+      closeDropdown();
+
+      setTimeout(function() {
+        this._input.focus();
+      }.bind(this), 0);
+    };
+
+    var getPreviousSiblingElement = element => {
+        if (!element) {
+          return null;
+        }
+        do {
+          element = element.previousSibling;
+        } while (element && element.nodeType != 1);
+        return element;
+    };
+
+    var getNextSiblingElement = element => {
+        if (!element) {
+          return null;
+        }
+        do {
+          element = element.nextSibling;
+        } while (element && element.nodeType != 1);
+        return element;
+    };
+
+    var setupListenerKeys = event => {
+      var activeItem = element.querySelector('.js-search-bar-dropdown-menu-item--active');
+
+      switch (event.keyCode) {
+      case 13:
+        // enter key
```

Comment:
Instead of comments I think these would be better as constants at the top of the function: `const ENTER_KEY_CODE = 13;`, then the code is nicely self-documenting.